tests/scr/testSysboxCI

```diff
@@ -56,6 +85,109 @@ else
+	set_sysbox_idmapped_mount "disable"
+
+	printf "\nExecuting Sysbox tests (shiftfs)\n"
+
+	printf "\nExecuting sysbox-mgr tests ... \n"
+	bats --tap tests/sysmgr
+	printf "\nExecuting sysbox-fs tests ... \n"
+	bats --tap tests/sysfs
+	printf "\nExecuting docker tests ... \n"
+	bats --tap tests/docker
+	printf "\nExecuting dind tests ... \n"
+	bats --tap tests/dind
+	printf "\nExecuting cind tests ... \n"
+	bats --tap tests/cind
+	printf "\nExecuting app tests (l1) ... \n"
+	bats --tap tests/apps/l1
+	printf "\nExecuting chown syscall tests ... \n"
+	bats --tap tests/syscall/chown
+	printf "\nExecuting basic mount syscall-interception tests ... \n"
+	bats --tap tests/syscall/mount/mount.bats
+	printf "\nExecuting mount syscall-interception tests for procfs resources ... \n"
+	bats --tap tests/syscall/mount/mount-procfs.bats
+	printf "\nExecuting mount syscall-interception tests for sysfs resources ... \n"
+	bats --tap tests/syscall/mount/mount-sysfs.bats
+	printf "\nExecuting mount syscall-interception tests for overlayfs resources ... \n"
+	bats --tap tests/syscall/mount/mount-overlayfs.bats
+	printf "\nExecuting mount syscall-interception tests for immutable resources ... \n"
+	bats --tap tests/syscall/mount/mount-immutables.bats
+
+	# Launch kind test-suite.
+	./tests/kind/kind.sh tests/kind/kind-custom-net.bats
+
+	# Launch sysbox-pod test-suite.
+	./tests/pods/pods.sh
+
+	printf "\nSysbox health checking ...\n"
+	bats --tap tests/health/sysbox-health.bats
+
+	docker system prune -a -f
+
+	printf "\nRe-enabling Sysbox idmapped-mounts.\n"
+	set_sysbox_idmapped_mount "enable"
+
+	return 0
+}
+
+function run_docker_userns_remap_tests() {
+	printf "\nExecuting Sysbox tests (docker-userns-remap)\n"
+
+	printf "\nConfiguring Docker with userns-remap\n"
+	docker system prune -a -f
+	docker-cfg --userns-remap=enable
+
+	printf "\nExecuting sysbox-mgr tests ... \n"
+	bats --tap tests/sysmgr
+	printf "\nExecuting sysbox-fs tests ... \n"
+	bats --tap tests/sysfs
+	printf "\nExecuting docker tests ... \n"
+	bats --tap tests/docker
+	printf "\nExecuting dind tests ... \n"
+	bats --tap tests/dind
+	printf "\nExecuting cind tests ... \n"
+	bats --tap tests/cind
+
+	# Launch basic kind-flannel test
+	docker system prune -a -f
+	printf "\nExecuting kind testcases with flannel cni ... \n"
+	bats --tap tests/kind/kind-flannel.bats
+	docker system prune -a -f
+
+	# Launch sysbox-pod test-suite.
+	./tests/pods/pods.sh
+
+	printf "\nSysbox health checking ...\n"
+	bats --tap tests/health/sysbox-health.bats
+
+	# TODO: disable all uid shifting in sysbox and verify things still work
+
+	printf "\nReverting Docker userns-remap config\n"
+	docker system prune -a -f
+	docker-cfg --userns-remap=disable
+}
+
+function main() {
+
+	# argument testName is optional; if present, only that test is executed.
+	if [ $# -eq 1 ]; then
+		printf "\nExecuting $1 ... \n"
+		bats --tap $1
+	else
+		run_idmapped_mount_tests
+		run_shiftfs_tests
+		run_docker_userns_remap_tests
```

Comment:
I don't think we should leave it out of the CI because some folks (notably Coder) use Docker in userns-remap mode. Plus it won't help a lot, the bulk of the test time is really in testing the idmapped + shiftfs, shiftfs-only, and idmapped-only scenarios.

I think it's best to work on making the whole test suite much faster per my prior comment.

Reply:
Let's talk about it.